File: android/src/main/java/com/cactuslab/capacitor/nativenavigation/types/ComponentType.java
package com.cactuslab.capacitor.nativenavigation.types;

import androidx.annotation.Nullable;

public enum ComponentType {
    STACK("stack"),
    TABS("tabs"),
    VIEW("view");

    private final String id;

    ComponentType(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    @Nullable
    public static ComponentType get(String id) {
        for (ComponentType value : values()) {
            if (value.id.contentEquals(id)) {
                return value;
            }
        }
        return null;
    }

}

File: android/src/main/java/com/cactuslab/capacitor/nativenavigation/types/ComponentType.kt
package com.cactuslab.capacitor.nativenavigation.types

enum class ComponentType(val id: String) {
    STACK("stack"), TABS("tabs"), VIEW("view");

    companion object {
        operator fun get(id: String?): ComponentType? {
            for (value in values()) {
                if (value.id.contentEquals(id)) {
                    return value
                }
            }
            return null
        }
    }
}